docs/source/notes/autograd.rst

```diff
@@ -849,20 +849,30 @@ Backward Hooks execution
 
 This section will discuss when different hooks fire or don't fire.
 Then it will discuss the order in which they are fired.
-The hooks that will be covered are: hooks registered to Tensor via
-:meth:`torch.tensor.register_hook`,
-post-hooks registered to Node via :meth:`torch.autograd.graph.Node.register_hook`, and
+The hooks that will be covered are: backward hooks registered to Tensor via
+:meth:`torch.Tensor.register_hook`, post-accumulate-grad hooks registered to
+Tensor via :meth:`torch.Tensor.register_post_accumulate_grad_hook`, post-hooks
+registered to Node via :meth:`torch.autograd.graph.Node.register_hook`, and
 pre-hooks registered to Node via :meth:`torch.autograd.graph.Node.register_prehook`.
 
 Whether a particular hook will be fired
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Hooks registered to a Tensor via :meth:`torch.tensor.register_hook`
+Hooks registered to a Tensor via :meth:`torch.Tensor.register_hook`
 are executed when gradients are being computed for that Tensor. (Note that this does not require
 the Tensor's grad_fn to be executed. For example, if the Tensor is passed
 as part of the ``inputs`` argument to :func:`torch.autograd.grad`,
 the Tensor's grad_fn may not be executed, but the hook register to that Tensor will always be executed.)
 
+Hooks registered to a Tensor via :meth:`torch.Tensor.register_post_accumulate_grad_hook`
+are executed after the gradients have been accumulated for that Tensor, meaning the
+Tensor's grad field has been set. Whereas hooks registered via :meth:`torch.Tensor.register_hook`
+are run as gradients are being computed, hooks registered via :meth:`torch.Tensor.register_post_accumulate_grad_hook`
+are only triggered once the Tensor's grad field is updated by autograd at the end of
+the backward pass. Thus, post-accumulate-grad hooks are only fired for leaf Tensors.
```

Comment:
Probably want to remove "Thus, post-accumulate-grad hooks are only fired for leaf Tensors." since you could not have registered them to non-leaf tensor anyway.